### PR TITLE
Document publishing workflow

### PR DESCRIPTION
**Pull request by idan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,3 +65,29 @@ heroku create
...
+   `npm install -g npm`
+2. Install `np` globally using npm: `npm install -g np`
+
+#### Release
+
+Because `yarn publish` doesn't support 2FA, we will have to invoke `np` with the
+`--no-yarn` option. This is a bit unfortunate because `np` is actually doing its
+sanity checks using `npm`, which might result in different behavior compared to
+`yarn`. Because of this, we should manually run the test suite ourselves ahead
+of `np` until such time as yarn gets unbroken.
```
Comment (uhoh-itsmaciek) [inline README.md]: Can we just turn off 2FA on publish (at least for now)? I don't believe we've been asked to turn this on, and I'd rather be more confident in our build artifacts than try to harden our workflow more than we have to.

Comment (idan) [reply README.md]: IDK, the whole point is that 2FA protects us from other people managing to publish code we use. If someone was looking to target us specifically, this seems like a good vector, but I guess it's @heroku/security 's call.

I don't think this has a significant impact on our confidence in build artifacts; we don't ship node_modules. I don't love that there's any discrepancy, but `¯\_(ツ)_/¯`

Comment (idan) [reply README.md]: Raised in #security: https://heroku.slack.com/archives/C1R3S6LVB/p1533058337000252

Comment (mattreduce) [reply README.md]: Thanks @idan! I appreciate the downside of having to use `npm` and the effects stemming from that, but our preference is to keep 2FA enabled. @uhoh-itsmaciek

Comment (uhoh-itsmaciek) [reply README.md]: @mattreduce just to be clear I wasn't arguing for turning off 2FA in general. NPM has three settings:

 * 2FA off
 * 2FA for login
 * 2FA for login *and* for publishing artifacts

I guess the preference is still for the third, though?

Comment (mattreduce) [reply README.md]: @uhoh-itsmaciek appreciate the clarification 👍 so for mode 2/2FA for login, we'd have to login to NPM first and could then push artifacts without 2FA, so no single request to push an artifact without 2FA could succeed without having first logged in and used a second factor?

Comment (mattreduce) [reply README.md]: @uhoh-itsmaciek yep. Okay, I'm following. Of course our favorite option is 3/require 2FA for login _and_ sensitive actions, but if going with 2 would alleviate these operational issues for `yarn` I think that's a good compromise. I found a Yarn issue about the lack of 2FA support and will keep an eye on that. Sound good to ya'll?

Comment (uhoh-itsmaciek) [reply README.md]: @mattreduce cool, sounds good. I think @idan's probably right and given the fact this does not affect build artifacts, his existing instructions are okay.

Comment (mattreduce) [reply README.md]: Right on. Thanks again for talking this through with me, ya'll 😄

Comment (idan) [reply README.md]: Thanks for clarifying @uhoh-itsmaciek, I did a crap job of explaining the multiple 2FA options on slack. 

I can imagine a situation where the generated artifacts could be different, but it's a stretch. If `npm` installed a different typescript for some reason, then the different `tsc` would build a different artifact.

I don't believe that `npm` and `yarn` would have such a digression in their behaviors, so I'm comfortable shipping this now, but if we ever encounter some funkiness in builds I'd look at this first.

**Pull request merged**